returnn/tf/util/data.py

```diff
@@ -5607,9 +5607,15 @@ def _create_size_placeholder(name, axis_wo_b, tag, batch_dim):
   """
   from .basic import reuse_name_scope
   with reuse_name_scope("extern_data/placeholders/%s" % name, absolute=True):
+    if batch_dim is not None and batch_dim.batch is not None:
+      batch = batch_dim.batch
+    elif tag.batch is not None:
+      batch = tag.batch
+    else:
+      batch = None
     dyn_size_ext = Data(
       "%s_dim%i_size" % (name, axis_wo_b), dtype=Data.size_dtype,
-      dim_tags=[batch_dim] if batch_dim else [], batch=batch_dim.batch if batch_dim else None)
+      dim_tags=[batch_dim] if batch_dim else [], batch=batch)
```

Comment:
I wonder, wouldn't that make more sense:
```suggestion
      dim_tags=[batch_dim] if batch_dim else [],
      batch=batch if batch_dim else None)
```